### ISSUE TITLE
projected pattern from projector in gazebo hydro appears backwards

### ISSUE THREAD

**Issue by hsu** [PR2/pr2_common#227]
We might need a fix in [urdf](https://github.com/PR2/pr2_common/blob/hydro-devel/pr2_description/urdf/sensors/projector_wg6802418.gazebo.xacro) or there could be a bug in gazebo.

**Comment by ahendrix** [comment]
This appears unaffected by the pose tag. I'm going to poke and see if it's a bug in the sdformat urdf parser.

**Comment by ahendrix** [comment]
Submitted a pull request. Not working yet, but getting closer: https://bitbucket.org/osrf/sdformat/pull-request/96/add-rpyxyz-parser-for-projector-pose/diff

**Comment by furushchev** [comment]
@hsu @ahendrix Is there any proceedings? I've reproduced this problem on indigo.

![screenshot from 2015-12-10 15 27 14](https://cloud.githubusercontent.com/assets/1901008/11708423/8e639284-9f52-11e5-8d9b-14067ce9ee81.png)

**Comment by aginika** [comment]
ping

**Comment by furushchev** [comment]
@hsu @ahendrix ping

**Comment by furushchev** [comment]
@hsu @ahendrix ping
or no maintainer of this package now?

**Comment by trainman419** [comment]
I'm no longer working on this.

**Comment by furushchev** [comment]
I fixed gazebo and sent PR.
https://bitbucket.org/osrf/gazebo/pull-requests/2152/fix-remove-back-projection/diff#comment-None

**Comment by furushchev** [comment]
PR is merged.

**Comment by 23pointsNorth** [comment]
Bump. Issue with projector still not fixed in indigo version.

**Comment by furushchev** [comment]
@23pointsNorth Sorry for very late response. And yes, my pull request that fixes this issue was merged into gazebo5, though gazebo version that we can install by `apt` on indigo is 2.2.3., where my patch is not yet applied.

**Comment by furushchev** [comment]
This problem appears again in gazebo7

![pr2_gazebo7](https://user-images.githubusercontent.com/1901008/38199040-19d38d04-36ca-11e8-8f45-fce555328c02.png)

Created an issue in bitbucket https://bitbucket.org/osrf/gazebo/issues/2449/back-projection-fixed-in-gazebo5-still

**Comment by furushchev** [comment]
From the discussion in https://bitbucket.org/osrf/gazebo/issues/2449/back-projection-fixed-in-gazebo5-still,
my fix for gazebo5 was ported into gazebo 7.1 (not gazebo 7.0).
To resolve this problem, we just have to upgrade gazebo following http://gazebosim.org/tutorials?cat=install&tut=install_ubuntu&ver=7.0#Alternativeinstallation:step-by-step

**Comment by v4hn** [comment]
Thank you for your efforts @furushchev this bug is pretty stupid to begin with.
I really don't understand how they failed to have their patches in 7.0 when they were released in previous versions...

**Comment by furushchev** [comment]
@v4hn True. They said that it's very hard to get fixes into the upstream Ubuntu gazebo7 package if it's not fixing a security vulnerability. 
If you (including me) still want to support me, please leave a comment to the original issue (or create a new issue) to tell them the importance of patching the bugfix. :+1:

**Comment by v4hn** [comment]
It's not that simple. Gazebo is an upstream Ubuntu package by now.
And 7.0 was released in a stable ubuntu distribution.
[SRUs are very hard to get in there](https://wiki.ubuntu.com/StableReleaseUpdates) and Gazebo 7.0->7.1 is a *much* bigger (and less important) change than what they consider worthy an SRU.

The Gazebo devs failed here. Now we have to live with it or use OSRF's custom debs, just as they proposed.
